font-types/src/tag.rs

```diff
@@ -259,9 +259,22 @@ impl<'de> serde::Deserialize<'de> for Tag {
     where
         D: serde::Deserializer<'de>,
     {
+        struct TagStrVisitor;
```

Comment:
noob question: any particular reason why this struct def/impl section is inlined in the `deserialized` function body and not at the top level as usual? Is it because it's only used in here? I presume there aren't any downsides to this (struct is defined only once, not each time one calls `deserialize`).

Reply:
correct, this is just a pattern you see sometimes, where small helper functions or types are just defined in the function body. It makes it clear that they are only referenced here.

Reply:
(and also correct, this compiles exactly the same as if it were declared at the top level; the only practical difference is that this type is not visible outside of this function)